test/integration/framework/utils.go

```diff
@@ -0,0 +1,307 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+
+package framework
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	kubernetesutils "github.com/gardener/gardener/pkg/utils/kubernetes"
+
+	"github.com/gardener/gardener/pkg/apis/garden/v1beta1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	kubecfg    = "kubecfg"
+	kubeconfig = "kubeconfig"
+	password   = "password"
+)
+
+// InitializeTestSeedClients will use the Garden Kubernetes client to read the Seed Secret in the Garden
+// cluster which contains a Kubeconfig that can be used to authenticate against the Seed cluster. With it,
+// a Kubernetes client for the Seed cluster will be initialized and attached to
+// the already existing GardenTestOperation object.
+func (o *GardenerTestOperation) initializeTestSeedClients() error {
+	if o.K8sSeedClient != nil {
+		return nil
+	}
+
+	k8sSeedClient, err := kubernetes.NewClientFromSecretObject(o.Seed.Secret, client.Options{})
+	if err != nil {
+		return err
+	}
+
+	o.K8sSeedClient = k8sSeedClient
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// initializeTestShootClients will use the Seed Kubernetes client to read the gardener Secret in the Seed
+// cluster which contains a Kubeconfig that can be used to authenticate against the Shoot cluster. With it,
+// a Kubernetes client for the Shoot cluster will be initialized and attached to
+// the already existing Operation object.
+func (o *GardenerTestOperation) initializeTestShootClients() error {
+	if o.K8sShootClient != nil {
+		return nil
+	}
+
+	k8sShootClient, err := kubernetes.NewClientFromSecret(o.K8sSeedClient, o.Shoot.SeedNamespace, v1beta1.GardenerName, client.Options{})
+	if err != nil {
+		return err
+	}
+
+	o.K8sShootClient = k8sShootClient
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *GardenerTestOperation) generateDeploymentObject(opts *TillerOptions) (*appsv1.Deployment, error) {
+	deploymentLabels := map[string]string{"name": "tiller"}
+	nodeSelectors := map[string]string{}
+	if len(opts.NodeSelectors) > 0 {
+		err := parseSelectorsIntoMap(opts.NodeSelectors, nodeSelectors)
```

Comment:
you should first parse the `nodeSelectors` and then just assign `nodeSelectors = parsedNodeSelectors`.